### PR TITLE
Customization of wave function for `TileReflection`

### PR DESCRIPTION
**Pull request by Lehonti**
The sharp, clear-cut edges of the tiles in `TileReflection` are achieved through the tangent function, which approaches infinite on one side and minus infinite on the other side, resulting in very sudden jumps.

But what if we chose a different function? For example if we chose sine, the image would be bulged in a tile pattern but it wouldn't have clear edges. Other functions would behave in different, interesting ways.

### CODE REVIEW

**Comment by cameronwhite** [comment]
This is cool, definitely has some interesting effects!

I'm mostly unsure about the naming of the `Wave Type` parameter and its menu options, which are pretty technical and might not make much sense to a user without an understanding of what the effect is doing internally. Ideally there could be some names that describe what behaviour is produced?

**Comment by Lehonti** [comment]
@cameronwhite what names would you suggest?

**Comment by cameronwhite** [comment]
I haven't thought much about names for the new modes, but for example you might label the existing mode as `Tile Shape` -> `Curved Edges` or something like that?
The current mode looks like the image is reflected in a mirror made up of tiles that have some curvature along their edges. Ideally we could have some labels for the new modes that give some idea of the general look of the effect

**Comment by Lehonti** [comment]
@cameronwhite if we focus on the visual effect and not the wave function as such, we'd have to think about this differently, so I reduced the number of options to two to start with (with more tile types coming in the future). Also, I gave the `enum` the name of `TileType` and the values are `CleavedEdges` (which is what we already have) and `CurvedEdges` (the new option). English is not my first language, so if you can think of better names, that would be cool.

File: Pinta.Effects/Effects/TileEffect.cs [hunk 1]
```diff
@@ -196,5 +218,17 @@ public sealed class TileData : EffectData
...
+		public TileType WaveType { get; set; } = TileType.CleavedEdges;
 	}
 }
+
+public enum TileType
+{
+	[Caption ("Cleaved Edges")]
+	CleavedEdges,
+
+	[Caption ("Curved Edges")]
```
Comment (cameronwhite) [inline Pinta.Effects/Effects/TileEffect.cs:232]: How about just `Curved` or `Round`?
From what I recall when testing it out last week, in this mode it's more like the whole tile is curved, whereas the other mode is more like a flat tile except at the edges

**Comment by Lehonti** [comment]
@cameronwhite  I implemented your suggestions 👍

**Pull request merged**